src/Umbraco.Web/Mvc/UmbracoViewPageOfTModel.cs

```diff
@@ -235,6 +236,18 @@ public override void WriteLiteral(object value)
             base.WriteLiteral(value);
         }
 
+        public override void Write(object value)
+        {
+            if (value is IHtmlEncodedString)
```

Comment:
I guess the issue with our own abstraction on this is that any other type of view isn't going to deal with this correctly. For example, in partial views or just normal views that aren't using this one then strange things will happen. I realize MS didn't make this hugely flexible since the underlying logic for all views uses `HttpUtility.HtmlEncode` which explicilty checks for `IHtmlString`. 

Not really sure the best answer to that apart from configuring the common base razor view type in each web.config in the views folder to be this one (i.e. https://haacked.com/archive/2011/02/21/changing-base-type-of-a-razor-view.aspx/) . Maybe there's another way to do that programatically with some sort of extension point for views.

Reply:
I'm not sure this is a real problem? 
Our views is generated with the `@inherits Umbraco.Web.Mvc.UmbracoViewPage<ContentModels.MyDocType>` in the top. 

You need this to use modelsbuilder right?

Reply:
Our normal template views are but that's only if  you create templates via the back office. You don't need to use that for models builder, the base class is just a helper. You can just as easily do `@model ContentModels.MyDocType`

Similarly this would be the approach for many people's partial views since most people aren't creating partial views via the back office but in VS instead. Many people will opt to using the base view class but it's not a requirement.